assets/js/script.js

```diff
@@ -103,8 +103,80 @@ function renderCard(eventObj) {
   
 function handleMoreInfoButtonClick(event, eventObj) {
     //TODO: Implement opening modal
+    const modal = document.getElementById('eventModal');
+    const modalContentEl = modal.querySelector('.box');
+    
+    // Clear previous content
+    modalContentEl.innerHTML = '';
+
+    // Title
+    const titleEl = document.createElement('h3');
```

Comment:
you don't need to re create the h3 (and the other) elements. you can just fetch it directly from the html and fill in the details. After which you can set the class for the modal to is-active.